tests/units/hindawi/test_hindawi_parser.py

```diff
@@ -0,0 +1,133 @@
+import xml.etree.ElementTree as ET
+
+from hindawi.parser import HindawiParser
+from pytest import fixture
+
+
+@fixture(scope="module")
+def hindawi_parser():
+    return HindawiParser()
+
+
+@fixture
+def articles(shared_datadir):
+    articles = []
+    with open(shared_datadir / "example1.xml") as file:
+        articles.append(ET.fromstring(file.read()))
+    return articles
+
+
+@fixture()
+def parsed_articles(hindawi_parser, articles):
+    return [hindawi_parser._publisher_specific_parsing(article) for article in articles]
+
+
+def test_dois(parsed_articles):
+    dois = ["10.1155/2019/3465159"]
+    for doi, article in zip(dois, parsed_articles):
+        assert "dois" in article
+        assert article["dois"] == [doi]
+
+
+def test_authors(parsed_articles):
+    authors = [
+        {
+            "affiliations": [
+                {
+                    "country": "Spain",
+                    "organization": "Grupo de Física Nuclear and Instituto "
+                    "Universitario de Física Fundamental y "
+                    "Matemáticas (IUFFyM), Universidad de "
+                    "Salamanca, E-37008 Salamanca",
+                    "value": "Grupo de Física Nuclear and Instituto "
+                    "Universitario de Física Fundamental y "
+                    "Matemáticas (IUFFyM), Universidad de Salamanca, "
+                    "E-37008 Salamanca, Spain",
+                }
+            ],
+            "orcid": "ORCID-0000-0003-2376-6255",
+            "raw_name": "Entem, David R.",
+        },
+    ]
+    for article in parsed_articles:
+        assert "authors" in article
+        assert article["authors"] == authors
+
+
+def test_abstracts(parsed_articles):
+    for article in parsed_articles:
+        assert "abstract" in article
+        assert (
+            article["abstract"]
+            == "In recent years, the discovery in quarkonium spectrum of several states not predicted by the naive quark model has awakened a lot of interest. A possible description of such states requires the enlargement of the quark model by introducing quark-antiquark pair creation or continuum coupling effects. The unquenching of the quark models is a way to take these new components into account. In the spirit of the Cornell Model, this is usually done by coupling perturbatively a quark-antiquark state with definite quantum numbers to the meson-meson channel with the closest threshold. In this work we present a method to coupled quark-antiquark states with meson-meson channels, taking into account effectively the nonperturbative coupling to all quark-antiquark states with the same quantum numbers. The method will be applied to the study of the X(3872) resonance and a comparison with the perturbative calculation will be performed."
```

Comment:
hmm pre-commit?

Reply:
☝️

Reply:
@ErnestaP could you please run pre-commit?

Reply:
It doesn't fix the issue :(